File: 1427-perform-string-shifts/1427-perform-string-shifts.cpp
class Solution {
public:
    string stringShift(string s, vector<vector<int>>& shift) {
        string answer = ""; 
        for(vector<int>& v : shift){
            int dir = v[0], amt = v[1] % s.length(); 
            if(dir == 0){
                s = s.substr(amt) + s.substr(0,amt); 
                //cout << s << ' '; 
            } else{
                s = s.substr(s.length() - amt) + s.substr(0,s.length() - amt); 
            }  
        }

        return s; 
    }
};

File: 1427-perform-string-shifts/1427-perform-string-shifts.java
class Solution {
    public String stringShift(String s, int[][] shift) {
        StringBuilder sb = new StringBuilder(s); 
        for(int[] p : shift){
            int dir = p[0]; 
            int len = p[1] % s.length(); 

            if(dir == 0){
                String left = sb.substring(0, len); 
                String right = sb.substring(len); 
                sb = new StringBuilder(right + left); 
            } else{
                String left = sb.substring(sb.length() - len); 
                String right = sb.substring(0, sb.length() - len); 
                sb = new StringBuilder(left + right); 
                //System.out.print(sb.toString()); 
            }
        }

        return sb.toString();
    }
}